src/setup/serverless-config.go

```diff
@@ -1,20 +1,114 @@
 package setup
 
-import log "github.com/sirupsen/logrus"
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+	"os"
+	"regexp"
+	"stellar/setup/deployment/connection/amazon"
+	"stellar/util"
+)
 
+// Serverless describes the serverless.yml contents.
 type Serverless struct {
-	// TODO: add serverless.yml fields
+	Service          string               `yaml:"service"`
+	FrameworkVersion string               `yaml:"frameworkVersion"`
+	Provider         Provider             `yaml:"provider"`
+	Package          Package              `yaml:"package"`
+	Functions        map[string]*Function `yaml:"functions"`
+}
+
+type Provider struct {
+	Name    string `yaml:"name"`
+	Runtime string `yaml:"runtime"`
+	Region  string `yaml:"region"`
+}
+
+type Package struct {
+	Patterns []string `yaml:"patterns"`
+}
```

Comment:
I think we can consider defining `Package` under the `Function` struct instead of the `Serverless` struct, so that the final generated `serverless.yml` looks a bit like the example [here](https://www.serverless.com/framework/docs/providers/aws/guide/packaging#packaging-functions-separately). Then we can enable the `package: individually: true` option to have the Serverless framework generate separate zip files for each function. I have not tested this though, so maybe we can also leave this for a separate PR.

Reply:
Thanks for bringing this up, I was not aware serverlesss can do that. I think it makes sense to explore this functionaality. I created a new isssue for this. Is it okay if I leave for another PR?

Reply:
Thank you for bringing this up. I was not aware serverless can do that and I think it's definitely worth exploring.  I created an issue in the project, is it ok if I leave it for a different PR?

Reply:
Yup that's fine! Thanks Maria